wordpress-mobile/WordPress-FluxC-Android
# ScanStore: Fix/ Ignore Threat

Parent wordpress-mobile/WordPress-Android#13326

This PR adds following endpoints [Internal Ref: pcdRpT-3J (Jetpack Scan API Audit)]

1. **Fix Threats** POST `/sites/$site/alerts/fix?threat_ids[1]=<id1>&threat_ids[2]=<id2>`
2. **Ignore Threat** POST `/sites/$site/alerts/$threat_id?ignore=true`
3. **Fetch Fix Threats Status** GET `/sites/$site/alerts/fix?threat_ids[1]=<id1>&threat_ids[2]=<id2>`

and adds corresponding `ScanAction`, models and methods in the `ScanRestClient` and `ScanThreatStore`.

To test

That newly added tests in `ScanRestClientTest`, `ScanStoreTests` run fine.

Notes

1. "Fix Threats" POST request is asynchronous. To check on the status of the fix, GET request is provided separately. 
2. To fix a single threat, we can pass corresponding threat id in the list to the "Fix Threats" API.
3. "Fix Threat Status" response is returned directly and not stored into the db. We'll consider adding to db if a need arises during implementation in the client app.
4. Connected tests not included yet (added to back log) as real threat ids from the test site might become invalid during internal testing. Will figure out a way those threats are not updated and then include tests.
5. For **Ignore Threat**, currently just checking if the request has succeeded and not worrying about the data in the response as it contains site's rewind site state data which doesn't seem to be useful, documented in calypso code [here](https://opengrok.a8c.com/source/xref/calypso/client/state/data-layer/wpcom/sites/alerts/ignore.js?r=bbad57e4#41) and commented in the Scan API Audit p2 post here (pcdRpT-3J#comment-201).
